src/sgame/sg_bot_ai.cpp

```diff
@@ -1050,6 +1050,103 @@ AINodeStatus_t BotActionResetStuckTime( gentity_t *self, AIGenericNode_t* )
 	return AINodeStatus_t::STATUS_SUCCESS;
 }
 
+// resupply action does 2 things:
+// * it makes the bot move at range of a resupply building
+// * it makes the bot to actively reload when at range to
+//   prevent them to go back to fight with one less clip...
+//   or to be stuck in the action.
+// TODO: handle aliens hugs to regen ammo (that's true for
+//   the BotActionHeal, mind you). This likely requires support
+//   for crowds.
+AINodeStatus_t BotActionResupply( gentity_t *self, AIGenericNode_t *node )
+{
+	team_t team = G_Team(self);
+	playerState_t const& ps = self->client->ps;
+	weaponAttributes_t const* wpa = BG_Weapon( BG_PrimaryWeapon( ps.stats ) );
+	ASSERT( self && self->botMind );
+	botMemory_t & mind = *self->botMind;
+
+	bool needsSupply = ps.ammo != wpa->maxAmmo
+		&& ps.clips != wpa->maxClips
+		&& wpa->maxAmmo != 0
+		&& !wpa->infiniteAmmo
+		&& ( team == TEAM_ALIENS && ( ps.stats[ STAT_STATE ] & SS_POISONED ) );
```

Comment:
You can only "need supply" when on aliens?

Reply:
barbs. Also, I'm trying to make the code as generic as possible to anticipate future changes.

Reply:
*Only* barbs? Never any human weapons?

Reply:
huh... oh, right, misunderstood your point. Indeed, bug spotted. Thanks.